api/src/main/java/com/okta/idx/sdk/api/client/BaseIDXClient.java

```diff
@@ -502,12 +505,40 @@ private void handleErrorResponse(Request request, Response response) throws IOEx
                 response.getHeaders().getContentType().toString().contains("application/ion+json")) {
             errorResponseJson = objectMapper.readTree(response.getBody());
             ErrorResponse errorResponseDetails = objectMapper.convertValue(errorResponseJson, ErrorResponse.class);
+            if (errorResponseDetails.getError() == null && errorResponseDetails.getMessages() == null) {
+                getMessagesFromRemediationOptions(errorResponseDetails, errorResponseJson);
+            }
             throw new ProcessingException(httpStatus, errorMsg, errorResponseDetails);
         } else {
             throw new ProcessingException(httpStatus, errorMsg);
         }
     }
 
+    private void getMessagesFromRemediationOptions(ErrorResponse errorResponseDetails, JsonNode errorResponseJson) {
+
+        IDXResponse idxResponse = objectMapper.convertValue(errorResponseJson, IDXResponse.class);
+        if(idxResponse != null && idxResponse.remediation() != null) {
+            for (RemediationOption remediationOption : idxResponse.remediation().remediationOptions()) {
+                if(remediationOption != null) {
+                    for (FormValue formValue : remediationOption.form()) {
+                        if(formValue != null && formValue.form() != null) {
+                            for (FormValue messageFormValue : formValue.form().getValue()) {
+                                if (messageFormValue.messages != null) {
+                                    ClassUtil.setInternalState(
```

Comment:
why do we have to do this? can we not expose a setter for `messages` instead of doing this the hacky way?